saleor/graphql/schema.graphql

```diff
@@ -3858,9 +3858,9 @@ type Product implements Node & ObjectWithMetadata {
   taxType: TaxType
   attributes: [SelectedAttribute!]!
   channelListings: [ProductChannelListing!]
-  imageById(id: ID): ProductImage
+  mediaById(id: ID): ProductMedia
```

Comment:
Theoretically, we are able to leave the old fields and deprecate them. I mean, we can have the same resolver which will always return only the images.
This change will break all custom storefronts.
@maarcingebala  what do you think? Is it possible?

Reply:
`imageById` is also public (not restricted with staff permission) so we cannot remove it.